File: 프로그래머스/0/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.py
def solution(arr, queries):
    result = []
    
    for s, e, k in queries:
        newArr = []
        for i in range(s, e+1):
            if arr[i] > k:
                newArr.append(arr[i])

        if newArr:
            newArr.sort()
            result.append(newArr[0])
        else:
            result.append(-1)
                
    return result               
            

File: 프로그래머스/0/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.js
function solution(arr, queries) {
    let result = [];
    
    for (let [s,e,k] of queries) {
        let min = Infinity;
        
        for (let j=s; j<=e; j++) {
            if(arr[j] > k && arr[j] < min) {
                min = arr[j];
            }
        }
        result.push(min == Infinity ? -1 : min);
    }
    return result;
}